### PR TITLE
refactored and removed npm dependency

### PR DESCRIPTION
**Pull request by matt-oc**
#23 
#22 
Also small update to docker file

### CODE REVIEW

**Comment by matt-oc** [comment]
@AdrianRossouw @mcdonnelldean Can ye review please

File: github.js [hunk 1]
```diff
@@ -2,100 +2,141 @@
 
 var GitHub = require('github')
 var github = new GitHub({version: '3.0.0'})
+var Request = require('request')
+var _ = require('lodash')
+var gitUrl
 
 var opts = {
-  token: ''
+  token: 'e414d503a227bb1445acf5ed619ec0948260babe'
```
Comment (mcdonnelldean) [inline github.js]: Remove your token :D

File: srv/github-dev.js [hunk 2]
```diff
@@ -2,47 +2,28 @@
...
-      if (mod) {
-        return this.act('role:info,res:part,part:github', {name: args.name, data: mod.data$()})
-      }
-
-      this.act('role:npm, cmd:get', {name: args.name}, function (err, mod) {
-        if (err) {
-          this.log.error(err)
-          return
-        }
+var TOKEN = process.env.TOKEN || 'e414d503a227bb1445acf5ed619ec0948260babe'
```
Comment (mcdonnelldean) [inline srv/github-dev.js]: Remove token

**Comment by matt-oc** [comment]
@mcdonnelldean Already done

**Comment by mcdonnelldean** [comment]
@matt-oc Is it tested?

**Comment by matt-oc** [comment]
@mcdonnelldean yup works perfectly, wouldnt PR anything untested.

**Comment by mcdonnelldean** [comment]
Awesome stuff!

**Pull request merged**